src/trace.rs

```diff
@@ -39,3 +53,137 @@ pub fn init(color: bool, json: bool, levels: &str) {
 pub fn current_span() -> Span {
     Span::current()
 }
+
+pub fn subscribe() -> Option<Receiver<Event>> {
+    SENDER.get().map(|sender| sender.subscribe())
+}
+
+struct BroadcastSubscriber<F> {
+    sender: Sender<Event>,
+    formatter: F,
+}
+
+impl<F: Subscriber + 'static> Subscriber for BroadcastSubscriber<F> {
+    #[inline]
+    fn enabled(&self, metadata: &Metadata<'_>) -> bool {
+        self.formatter.enabled(metadata)
+    }
+
+    #[inline]
+    fn new_span(&self, span: &tracing::span::Attributes<'_>) -> Id {
+        self.formatter.new_span(span)
+    }
+
+    #[inline]
+    fn record(&self, span: &Id, record: &tracing::span::Record<'_>) {
+        self.formatter.record(span, record)
+    }
+
+    #[inline]
+    fn record_follows_from(&self, span: &Id, follows: &Id) {
+        self.formatter.record_follows_from(span, follows)
+    }
+
+    #[inline]
+    fn event(&self, event: &tracing::Event<'_>) {
+        if self.sender.receiver_count() > 0 {
+            let _ = self.sender.send(event.into()); // Ignore errors
```

Comment:
This might drop the events if the `sender` channel is full, right? It the desired behaviour imo, just asking if it does indeed do that.

Reply:
Yes, this is a "best effort" kind of send. This is effectively required behavior, because the alternative (blocking if the channel is full) could quite trivially cause vector to lock up. Reporting the error could similarly lead to an infinite loop.

However, while investigating this behavior, I see that *receiving* messages when the channel is full can produce a temporary error, which should not cause the `internal_logs` source to stop (which it will now). I'll have to fix that.